### DISCUSSION TITLE
Inter var but not actually Inter var?

### DISCUSSION THREAD

**Discussion by geoffreymcgill** [tailwindlabs/tailwindcss#10909] [Help] [answered]
The font used on [tailwindcss.com/docs](https://tailwindcss.com/docs) is `Inter var`. 

<img width="115" alt="Screen Shot 2023-03-29 at 10 24 04 PM" src="https://user-images.githubusercontent.com/62210/228730988-08200c6b-de4b-47be-b6b0-96cd06141e0a.png">

But it appears to be a different font than what is available on the actual `Inter var` website ([rsms.me/inter](https://rsms.me/inter)).

<img width="109" alt="Screen Shot 2023-03-29 at 10 24 13 PM" src="https://user-images.githubusercontent.com/62210/228731190-71433fd8-033c-4e47-8140-3eff49b8f3b5.png">

Notice the difference in the `a` glyph. 

I must be missing something completely obvious here, but why is a different `a` glyph used on the [tailwind.css/docs](https://tailwindcss.com/docs) site vs [rsms.me/inter](https://rsms.me/inter)? I'm sure other characters are different too, just the `a` is an obvious one.

**Comment by geoffreymcgill** [comment]
I can download the actual [`Inter-roman-latin.var.18496762.woff2`](https://tailwindcss.com/_next/static/media/Inter-roman-latin.var.18496762.woff2) from the tailwindcss.com/docs website and use font file my project, and I get the `a` glyph as spec'd on the [rsms.me/inter](https://rsms.me/inter) website.

So, I'm at a loss as to what font is actually being used on the tailwindcss.com/docs website or what is causing unique glyphs, such as the `a`.

<img width="368" alt="Screen Shot 2023-03-29 at 10 31 03 PM" src="https://user-images.githubusercontent.com/62210/228733311-e3a308dd-bb71-4fc7-8776-5e2353d46c30.png">

Any help is greatly appreciated.

**Comment by adamwathan** [comment, marked as answer]
We set `font-feature-settings: "cv11"` to enable the single story `a` alternate glyph:

https://rsms.me/inter/lab/?feat-cv11=1

Inter includes lots of alternate styles like this that you can access as OpenType font features 👍

**Reply by geoffreymcgill** [reply]
Interesting. Learned something new today. 

https://developer.mozilla.org/en-US/docs/Web/CSS/font-feature-settings

```css
html {
    font-feature-settings: "cv02", "cv03", "cv04", "cv11";
}
```

**Reply by rruslianto14** [reply]
Hi just wondering how do you set the font-feature-setting in tailwind on a next.js app? I have am using the Google inter font but can’t find a way to set it to “cv11” just wondering if you found a way how? Thanks in advance.

**Reply by wongjn** [reply]
You could add the CSS from https://github.com/tailwindlabs/tailwindcss/discussions/10909#discussioncomment-5473347 into a CSS file that gets used in your Next.js app. You could also [setting the `fontFeatureSettings` when customizing the `fontFamily` configuration](https://tailwindcss.com/docs/font-family#providing-default-font-settings) in the Tailwind configuration.

**Reply by rruslianto14** [reply]
@wongjn thanks for your reply. I have tried both and still can't get the 'cv11' version of the 'a' character to render correctly. I am using the Inter font from Google and next/font so not sure if I need to do something else? Appreciate your help in advance! 

**layout.tsx**
```typescript
import { Inter } from "next/font/google";
import "./globals.css";

const inter = Inter({
  subsets: ['latin'],
  variable: '--font-inter',
})

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="en">
      <body>
        <div className={`${inter.variable} font-sans`}>
          {children}
        </div>
      </body>
    </html>
  );
}

``` 

**page.tsx**
```typescript
export default function Home() {
  return (
    <>
      <header>
        <div>
          <p>a test</p>
        </div>
      </header>
    </>
  );
}
``` 

**tailwind.config.ts**
```typescript
import type { Config } from "tailwindcss";
const defaultTheme = require("tailwindcss/defaultTheme");

const config: Config = {
  content: [
    "./src/pages/**/*.{js,ts,jsx,tsx,mdx}",
    "./src/components/**/*.{js,ts,jsx,tsx,mdx}",
    "./src/app/**/*.{js,ts,jsx,tsx,mdx}",
  ],
  theme: {
    extend: {
      fontFamily: {
        sans: [
          "var(--font-inter)",
          {
            fontFeatureSettings: '"cv11"',
            fontVariationSettings: "normal",
          },
        ],
      },
    },
  },
  plugins: [],
};
export default config;
``` 

**global.css**
```
@tailwind base;
@tailwind components;
@tailwind utilities;

html {
  height: 100%;
  font-feature-settings: 'cv11';
}
```

**Reply by wongjn** [reply]
You seem to be using Google Font's version of the Inter font. If [we look up the axes that Google Font's Inter supports](https://fonts.google.com/variablefonts?vfquery=Inter), we see only `slnt` and `wght` for slant and weight respectively. To support `cv11`, you could consider [downloading from the author's site](https://rsms.me/inter/download/), which supports more axes, including `cv11` for the alternative `a` glyph. See [this StackBlitz project](https://stackblitz.com/edit/stackblitz-starters-gsbljq?description=The%20React%20framework%20for%20production&file=src%2Fapp%2Flayout.tsx&title=Next.js%20Starter) that compares the two different font files.

**Reply by rruslianto14** [reply]
Thanks again for taking the time to explain and create the StackBlitz project @wongjn! 

I think I get it but just had a follow up question now; how can we use the **local font** to render the double-storey 'a' that is used in the Google font?  

In your StackBltiz project, I have commented out/removed the following in the tailwind.config.ts `fontFeatureSettings: '"cv11"` and `fontVariationSettings: 'normal'` for the **local font** and also the `font-feature-settings: 'cv11';` in the global.css file, however, I see that the local font is still rendering the single-storey 'a'? 

If my understanding is correct, shouldn't the local font now render the double-storey 'a' (same as the Google font one) as we have removed and haven't specified to use 'cv11'?

<img width="363" alt="image" src="https://github.com/tailwindlabs/tailwindcss/assets/57468211/b39d53ec-8f3d-4073-b155-7f2f3bf21215">
 

Thanks again in advance.    

**tailwind.config.ts**
```typescript
import type { Config } from 'tailwindcss';

const config: Config = {
  content: [
    './src/pages/**/*.{js,ts,jsx,tsx,mdx}',
    './src/components/**/*.{js,ts,jsx,tsx,mdx}',
    './src/app/**/*.{js,ts,jsx,tsx,mdx}',
  ],
  theme: {
    extend: {
      fontFamily: {
        sans: [
          'var(--font-inter-google)',
          {
            fontFeatureSettings: '"cv11"',
            fontVariationSettings: 'normal',
          },
        ],
        local: [
          'var(--font-inter-local)',
          // {
          //   fontFeatureSettings: '"cv11"',
          //   fontVariationSettings: "normal",
          // },
        ],
      },
    },
  },
  plugins: [],
};
export default config;
``` 

**global.css**
```
@tailwind base;
@tailwind components;
@tailwind utilities;

html {
  height: 100%;
  /* font-feature-settings: 'cv11'; */
}
```

**Reply by wongjn** [reply]
The styles from `font-sans` gets applied to the `<html>` element from preflight:

https://github.com/tailwindlabs/tailwindcss/blob/f1f419a9ecfcd00a2001ee96ab252739fca47564/src/css/preflight.css#L36-L38

Thus, even commenting out `fontFeatureSettings: '"cv11"'` from `fontFamily.local`, the `<div>` still inherits `font-feature-settings: "cv11"` from the `html` parent. To demonstrate this, [here is the same project with `cv11` removed for `fontFamily.local` and preflight disabled](https://stackblitz.com/edit/10909-9323689?file=tailwind.config.ts,src%2Fapp%2Flayout.tsx) and observe that the `a` characters are now the same style.

**Reply by rruslianto14** [reply]
@wongjn great! Thanks for taking the time to explain it, made a lot of sense and learned something new! Cheers!

**Comment by braianorl** [comment]
Is it possible to use this font on my Figma file? I tried download the Inter var but only the Italic came with the cv11

**Reply by braianorl** [reply]
I just found a solution for using on both figma and bubble.io: I download the InterVariable from the author's website. And then I download a software to freeze a specific feature of the opentype. That way I had exported a frozen Intervariable with the specific characteristics I like and could use both on bubble and figma

**Comment by rickyzhangca** [comment]
the easiest/fastest way is probably:

```
<!-- HTML in your document's head -->
<link rel="preconnect" href="https://rsms.me/">
<link rel="stylesheet" href="https://rsms.me/inter/inter.css">
```

```
/* CSS */
:root {
  font-family: Inter, sans-serif;
  font-feature-settings: 'liga' 1, 'calt' 1; 'cv11': 1;
}
@supports (font-variation-settings: normal) {
  :root { font-family: InterVariable, sans-serif; }
}
```

**Reply by developerdavid2** [reply]
Thanks, this really did work for me. I was having conflicting issues using PrimeReact with Tailwindcss especially as regards using the inter font and this simple solution solved it.

**Reply by bxavace** [reply]
Thanks! This fixed the weird font issues on my Angular 17 app with PrimeNG 17 and Tailwind 3.4.13. I thought I was the only one finding that weird change in the glyph!

**Comment by Sbrjt** [comment]
tldr:

Download it from [here](https://rsms.me/inter/download/). Google fonts lacks `cv11`.

In layout.tsx:

    const inter = localFont({
        src: [
            {
                path: '../../public/fonts/InterVariable.woff2',
                weight: '100 900',
                style: 'normal',
            },
            // ... also for italic (no need for static fonts, we only need InterVariable)
        ],
        variable: '--font-inter',
    })
    
    function RootLayout({ children }) {
	return (
		<html lang='en'>
			<body className={inter.variable}>
			     ....
			     
In global.css:

    // use this class for inter
    .font-inter {
        font-family: var(--font-inter);
        font-feature-settings: 'cv11' 1;
    }
